README.md

```diff
@@ -1 +1,1146 @@
-# eip-message-transformation
\ No newline at end of file
+[![Build Status](https://travis-ci.org/sanethmaduranga/eip-message-transformation.svg?branch=master)](https://travis-ci.org/sanethmaduranga/eip-message-transformation.svg?branch=master)
+
+# eip-message-transformation
+There are different ways of message transformation methods in EIP (Enterprise Integration Patterns). In this guide, we are focusing on *content filter*, *claim check* and *content enricher* message transformation methods between services using an example scenario.
+
+> This guide describes implementing three message transformation patterns using Ballerina programming language as simple steps.
+
+The following are the sections available in this guide.
+
+- [What you'll build](#what-youll-build)
+- [Prerequisites](#prerequisites)
+- [Implementation](#implementation)
+- [Testing](#testing)
+- [Deployment](#deployment)
+- [Observability](#observability)
+
+## What you’ll build
+When it comes to the data communication, the major challenge is formats of storage mechanisms vary among the systems. 
+
+![alt text](images/messagetrasformation.png)
+
+Also, the message producers and consumers use different techniques according to their requirement. So message transformation plays an important role in coupling those message producers and the message consumers. 
+Not only that, the performance impact while message transformation is also an important fact in the real world. Here we discuss main three message transformation patterns in Enterprise Integration as content filter, content enricher, and claim check.
+
+### Content filter
+The content filter EIP (Enterprise Integration Pattern) important when we need to manage a large message in order to get a few data from it. It removes unimportant data items from a message and leaves only the important ones. In addition to removing data elements, Content Filter can be used to simplify a message structure.
+
+![alt text](images/contentfilter.png)
+
+In our sample scenario, input request contains a lot of student's details. So content filter uses to simplify the input request such as request contains only student ID. Additional data such as student name, student's city, and gender will be dropped to ensure the performance of message transformation.
```

Comment:
Change to 'So, the content filter is used to simplify the input request so that the request contains only the student ID.'

Reply:
Done

Reply:
Cannot see this change somehow.